documentation/docs/30-advanced/60-images.md

```diff
@@ -0,0 +1,149 @@
+
+```js
+const pictures = import.meta.glob(
+	'/path/to/assets/*.{avif,gif,heif,jpeg,jpg,png,tiff,webp}',
+	{
+		query: {
+			enhancedImg: true
+		}
+	}
+);
+```
+
+### Intrinsic Dimensions
+
+`width` and `height` are optional as they can be inferred from the source image and will be automatically added when the `<enhanced:img>` tag is preprocessed. These attributes are added because the browser can reserve the correct amount of space when it knows image dimensions which prevents layout shift. If you'd like to use a different `width` and `height` you can style the image with CSS. Because the preprocessor adds a `width` and `height` for you, if you'd like one of the dimensions to be automatically calculated then you will need to specify that:
+
+```svelte
+<style>
+	.hero-image img {
+		width: var(--size);
+		height: auto;
+	}
+</style>
+```
+
+### `srcset` and `sizes`
+
+You may be wondering how to create something that looks like:
+
+```html
+<img
+  srcset="image-small.png 160w, image-medium.png 400w, image-large.png 600w"
+  sizes="(min-width: 60rem) 80vw, (min-width: 40rem) 90vw, 100vw"
+  />
+```
+
+In this example, it would be tedious to have to manually create three versions of your image. Instead, you can specify the widths as a query parameter and we'll generate the `srcset` for you.
+
+```svelte
+<enhanced:img
+  src="./image.png?w=160,400,600"
+  sizes="(min-width: 60rem) 80vw, (min-width: 40rem) 90vw, 100vw"
+  />
+```
+
+If `sizes` is specified directly as a string on the `img` tag then the plugin will automatically generate different width images:
+- If `sizes` is not provided, then a HiDPI/Retina image and a standard resolution image will be generated. The image you provide should be 2x the resolution you wish to display so that the browser can display that image on devices with a high [device pixel ratio](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio).
+- If `sizes` is provided, then a large `srcset` is generated. If some of the `sizes` have been specified as a percentage of the viewport width using the `vw` unit then the `srcset` will filter out any values which are too small to ever be requested by the browser.
+
+> Text expressions like `sizes={computedSizes}` or `sizes={['(min-width: 60rem) 80vw', '(min-width: 40rem) 90vw', '100vw'].join(', ')}` will not be evaluated for the purposes of automatic image generation and will be skipped.
+
+### Per-image transforms
+
+By default, enhanced images will be transformed to more efficient formats. However, you may wish to apply other transforms such as a blur, quality, flatten, or rotate operation. You can run per-image transforms by appending a query string:
+
+```svelte
+<enhanced:img src="./path/to/your/image.jpg?blur=15" alt="An alt text" />
+```
+
+[See the imagetools repo for the full list of directives](https://github.com/JonasKruckenberg/imagetools/blob/main/docs/directives.md).
+
```

Comment:
```suggestion

## Loading images from a CDN

In some cases, the images may not be accessible at build time — they live inside a content management system or elsewhere.

Using a content delivery network (CDN) can allow you to optimize these images dynamically, and provides more flexibility with regards to sizes, but it may involve some setup overhead and usage costs. Depending on caching strategy, the browser may not be able to use a cached copy of the asset until a [304 response](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/304) is received from the CDN. Building HTML to target CDNs may result in slightly smaller and simpler HTML because they can serve the appropriate file format for an `<img>` tag based on the `User-Agent` header whereas build-time optimizations must produce `<picture>` tags with multiple sources. Finally, some CDNs may generate images lazily, which could have a negative performance impact for sites with low traffic and frequently changing images. We do not currently offer any tools for dynamic image transforms, but we may offer such utilities in the future.

You can mix and match all three solutions in one project. For example, you may display images on your homepage with `@sveltejs/enhanced-img` and display user-submitted content with a dynamic approach.

```